homeassistant/bootstrap.py

```diff
@@ -187,6 +187,8 @@ async def async_setup_hass(
     if runtime_config.open_ui:
         hass.add_job(open_hass_ui, hass)
 
+    core._cv_hass.set(hass)  # pylint: disable=protected-access
```

Comment:
I have been testing with this. The `hass` is needed during the config validations in the platforms we need to set this right after setting the `hass` variable and not the end.

https://github.com/home-assistant/core/pull/76303/files#diff-7f269b55a61696a8682452bbeb39f29e4f7065f1a03cc990b8f46cd2efb9726dR102

Reply:
This is a concept, and might not be the path we take at all; another option we are discussing, for example, is making HomeAssistant a singleton.